craft_application/git/_git_repo.py

```diff
@@ -120,6 +138,24 @@ def parse_describe(describe_str: str) -> str:
     return f"{version}.post{distance}+git{commit[1:]}"
 
 
+@lru_cache(maxsize=1)
+def _get_git_binary_name() -> str:
```

Comment:
What's the benefit of putting this in a protected function that just gets called from that classmethod?

Reply:
Probably there is no benefit from doing that, I'll cache `classmethod` directly

Reply:
Changed and tested in https://github.com/canonical/craft-application/pull/576/commits/ee680832348052d0af9df8a4d9c2094129957d6e